binary-oracle-pair/program/src/processor.rs

```diff
@@ -255,9 +281,12 @@ impl Processor {
         pool.decider = *decider_info.key;
         pool.mint_end_slot = mint_end_slot;
         pool.decide_end_slot = decide_end_slot;
-        pool.decision = None;
+        pool.decision = Decision::Undecided;
+
+        pool.serialize(&mut *pool_account_info.data.borrow_mut())
+            .unwrap();
```

Comment:
nit: since the Borsh error can become a ProgramError, you can actually just do, `pool.serialize(&mut *pool_account_info.data.borrow_mut())` as the last line!

Reply:
When I'm trying to do this I have such an error
```
expected enum std::result::Result<_, solana_program::program_error::ProgramError>
         found enum std::result::Result<_, std::io::Error>
```

Reply:
ah right, you still need to map the error, like `pool.serialize(...).map_err(|e| e.into())`